deepchem/models/torch_models/layers.py

```diff
@@ -2972,3 +2972,339 @@ def _dynamic_partition(self, input_tensor: torch.Tensor,
         partitions = [input_tensor[mask] for mask in partition_masks]
 
         return partitions
+
+
+class WeaveLayer(nn.Module):
```

Comment:
Can you add this class to the layers.rst file in the docs? We should make sure it appears in the docs for layers

Reply:
I have now added WeaveLayer to docs.